apps/bridge-dapp/src/containers/UploadSpendNoteModal/UploadModalContent.tsx

```diff
@@ -58,6 +58,15 @@ export const UploadModalContent: FC<UploadModalContentProps> = ({
   // useMemo for note size
   const noteSize = useMemo(() => Object.keys(notes).length, [notes]);
 
+  // Parse JSON string
+  const parseJSON = (str: string) => {
```

Comment:
Please consider using `useCallback` here, also adding document for the return type of this function.